web_src/js/utils/dom.js

```diff
@@ -0,0 +1,65 @@
+function getComputedStyleProperty(el, prop) {
+  const cs = el ? window.getComputedStyle(el) : null;
+  return cs ? cs[prop] : null;
+}
+
+function isShown(el) {
+  return getComputedStyleProperty(el, 'display') !== 'none';
+}
+
+function assertShown(el, expectShown) {
+  if (window.config.runModeIsProd) return;
+
+  // to help developers to catch display bugs, this assertion can be removed after next release cycle or if it has been proved that there is no bug.
+  if (expectShown && !isShown(el)) {
+    throw new Error('element is hidden but should be shown');
+  } else if (!expectShown && isShown(el)) {
+    throw new Error('element is shown but should be hidden');
```

Comment:
```suggestion
    throw new Error(`element ${el} is hidden but should be shown`);
  } else if (!expectShown && isShown(el)) {
    throw new Error(`element ${el} is shown but should be hidden`);
```

Reply:
No, you may not know whether `el` could be `toString` correctly.

Reply:
But even if it doesn't override `toString()`, having this information is much more valuable than not having it:
If it overrides `toString`, we get an estimate **which** component behaves faulty.
If it doesn't override `toString`, we at least know that it is none of the browser native elements as they override `toString`.

Reply:
And this message is only for developers, when the error occurs, developers can see the stacktrace from console, that's more clear. So I think it's not necessary to convert the uncertain `el` to a string.

Reply:
I do not know why you insist this. When you see message:

`element [object HTMLBodyElement] is shown but should be hidden`

or 

`element [object HTMLDivElement] is shown but should be hidden`

It helps nothing, in the end, you have to get more details from your console, to see where what happens

Reply:
Okay, yeah, TIL I've been fooled by my browser console :sweat: :
I thought the `<span class="author flex-self-stretch" itemprop="author">` I see for example when entering `document.querySelector(".author")` in the console of this page was its `toString()` representation.
Apparently this was not the case, as `document.querySelector(".author").toString()` returns the undescriptive `"[object HTMLSpanElement]"` you mentioned.
Then it's not needed.
Sorry for the confusion.